portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase

```diff
@@ -22,242 +22,127 @@ definition {
-
-			Navigator.gotoBack();
-		}
+		task ("Authorize application2 and launch request") {
 
-		task ("Edit scopes") {
-			OAuth2.editScopes(
-				applicationName = "OAuth Application",
-				resourceCheckList = "liferay-json-web-services.read.portalversion",
-				resourcePanels = "Portal Services");
-		}
+			// Repeating the authorization on a second page
 
-		task ("Get access token") {
-			var accessToken = OAuth2.getAccessToken(
-				clientId = "${generatedClientId}",
-				clientSecret = "${generatedClientSecret}",
-				grantFlow = "clientCredentials");
-		}
+			OAuth2.authorizeMyApplication(
+				clientId = "default2",
+				clientSecret = "defaultsecret2",
+				newTab = "true",
+				rememberDevice = "true");
 
-		task ("Assert portal version equals 7.4") {
-			SystemSettings.assertPortalVersion(
-				accessToken = "${accessToken}",
-				portalVersion = "7.4");
+			OAuth2.launchRequest();
 		}
+		task ("Authorize application2 and launch request") {
 
-		task ("Edit SAP successfully") {
-			Navigator.openSpecificURL(url = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_portal_security_service_access_policy_web_portlet_SAPPortlet");
-
-			ServiceAccessPolicy.editServiceAccessPolicy(
-				enabled = "false",
-				policyName = "OAUTH2_read.portalversion");
-
-			var portalVersionOutput = JSONCurlUtil.post("${portalVersionCurl}");
-
-			OAuth2.checkIfMessageNotDenied(
-				msg = "Access denied to com.liferay.portal.kernel.service.PortalService#getVersion",
-				portalVersionOutput = "${portalVersionOutput}");
-		}
+			// Re-open test harness to check if Device is remembered
 
-		task ("View OAuth application scope") {
-			Navigator.openSpecificURL(url = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_oauth2_provider_web_internal_portlet_OAuth2AdminPortlet");
+			OAuth2.authorizeMyApplication(
+				clientId = "default2",
+				clientSecret = "defaultsecret2",
+				noAuthorize = "true");
 
-			OAuth2.viewScopes(
-				applicationName = "OAuth Application",
-				removedScope = "read.portalversion",
-				resourcePanels = "Portal Services");
+			OAuth2.launchRequest();
 		}
+	}
 
-		task ("Edit SAP") {
-			Navigator.openSpecificURL(url = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_portal_security_service_access_policy_web_portlet_SAPPortlet");
-
-			ServiceAccessPolicy.editServiceAccessPolicy(
-				enabled = "true",
-				policyName = "OAUTH2_read.portalversion");
-		}
+	@description = "This is a use case for LPS-126597. Authorize external application with remember device enabled."
+	@priority = "5"
+	test ApplicationCanBeAuthorizedWithRememberDeviceEnabled {
+		property test.name.skip.portal.instance = "OAuth2#ExternalApplicationCanBeAuthorizedAndRememberDeviceShouldBeEnabled";
 
-		task ("Assert portal version equals 7.4") {
-			SystemSettings.assertPortalVersion(
-				accessToken = "${accessToken}",
-				portalVersion = "7.4");
+		task ("Add and edit external application with remember device enabled") {
```

Comment:
I think if we use Add macro, then we don't have to mention that we also edit it. I expect that the editing happens during add, aka. OAuth2.addAndEditExternalApplication could be just OAuth2.addNewExternalApplication, and at the macro level we do decorate the application as we want (aka edit it)